### PR TITLE
v3.3.0

### PR DESCRIPTION
**Pull request by mxstbr**
This is a release candidate for `v3.3.0` of `react-boilerplate`! Please test it and let me know if something is broken, and we'll ship this as soon as a handful of people have confirmed everything works as expected!

This release directly contains the brainpower of 25 individual contributors, with many more chiming into discussions. Thank you so much for helping out, without you `react-boilerplate` wouldn't be as great as it is!

# Changelog

## Main

- **Use [💅 `styled-components`](https://github.com/styled-components/styled-components)**, huge thanks to @samit4me 
- **New, smaller translation messages format**, thanks @chaintng 
- **Fix sync reducer HMR**, thanks @justingreenberg 
- Add pure component support to the generator, thanks @justingreenberg 
- Add `eslint-plugin-redux-saga`, thanks @jstockwin 
- Remove `offline-plugin` warning in dev, thanks @sedubois 
- Show a warning if a user has JavaScript disabled, thanks @Vadorequest 
- Add yarn support, thanks @EugeneHlushko @sedubois 
- Add `NamedModulesPlugin`, thanks @simobasso 

## Other Updates

- Update dependencies, thanks to @gihrig 
- Fix container generator, thanks to @lsanders 
- Add a `key` to `LanguageProvider` to force rerenders, thanks @andregoncalvesdev
- Remove DevTools code in production, thanks @ansumanshah 
- Showcase `sCU` in `FeaturePage`, thanks @justingreenberg 
- Use `browser` field of modules that supply it, thanks @chaintng @episage 
- Fix `npm run analyze`, thanks @chaintng 
- Fix dependency warnings, thanks @sedubois 
- Migrate to `eslint-config-airbnb` v12, thanks @amilajack 
- Update webpack, thanks @peter-mouland 
- Update Redux DevTools, thanks @zalmoxisus 
- Update documentation with lots of info, thanks @Vadorequest @nappels @KarandikarMihir @GGAlanSmithee @malixsys @sethmcleod 
- Clean up script files, thanks @lvarayut 
- Make AppVeyor faster by caching `node_modules`, thanks @amilajack 
- Fix cleaned state, thanks @rubenmoya 
- Simplify example CSS, thanks @sedubois 
- Clean up example, thanks @sedubois @nktssh 
- Use `takeLatest` in example saga, thanks @pyjac 

# News

Welcome @chaintng and @sedubois to the team, they'll be around the repo closing issues and answering questions!  Special thanks as always to @gihrig for diligently testing everything and keeping our dependencies up to date!

If you want to help us make react-boilerplate great, please use the `dev` branch, test all the things and report all the bugs!

# Supporters

This release was made possible by [Fullstack React](https://www.fullstackreact.com/) and [Serverless](http://serverless.com)! _Want to support us too? [Click here!](http://opencollective.com/react-boilerplate)_

### CODE REVIEW

**Comment by mxstbr** [comment]
Deploy preview failed.

Built with commit 4cb38d9b8bd8e0b228ce9b31e67755393fb3eb0b

https://app.netlify.com/sites/react-boilerplate/deploys/5819abdcc4d9cc191a086933

**Comment by mxstbr** [comment]
Not sure why the deploy preview failed, seems to be a Netlify issue. Will investigate, meanwhile please try this branch and let me know if I missed something!

File: internals/webpack/webpack.base.babel.js [hunk 1]
```diff
@@ -70,31 +70,38 @@ module.exports = (options) => ({
...
     extensions: [
-      '',
       '.js',
       '.jsx',
       '.react.js',
     ],
     mainFields: [
+      'browser',
       'jsnext:main',
+      'browser',
```
Comment (ooflorent) [inline internals/webpack/webpack.base.babel.js]: Why is it added twice?

Comment (mxstbr) [inline internals/webpack/webpack.base.babel.js]: Arg god dammit, that was my mistake. Good catch, will fix!

Comment (mxstbr) [inline internals/webpack/webpack.base.babel.js]: Fixed in b9a229d006a7ad238639b0cd99f2cf3fa3f6ce1f

**Comment by nhducit** [comment]
How can I keep my project up to date with react-boilerplate? I'm using v3.2.1

**Comment by mxstbr** [comment]
Shipping this tomorrow morning assuming CI doesn't compain. /cc @gihrig @oliverturner @justingreenberg 

Honestly, I'm ridiculously excited about this release. 24 people have directly contributed some form of code or documentation for v3.3! Amazing stuff.

**Comment by jgrancher** [comment]
As @nhducit asked, would there be a relatively simple way for people using an older version of this boilerplate to update it? (I'm still working on a project that I had started with 3.2, and would be keen to use these cool new features 😃 )

**Comment by samit4me** [comment]
@nhducit @jgrancher Everyone will have their own ways of doing this but for me personally I add a remote called `upstream` pointing to this repo. Then whenever there is a new version of the `master` branch I will either Merge or Rebase the changes into my project. There is a rough idea on how to do this under the [CONTRIBUTING](https://github.com/mxstbr/react-boilerplate/blob/master/.github/CONTRIBUTING.md#pull-requests) file, but I just opened an issue #1180 so we can better document this issue. If you have any thoughts or recommendations, please put them on that issue, would like to hear them!

**Comment by spentay** [comment]
>  I add a remote called upstream pointing to this repo.

I think an idea would be to call the branch something like `boilerplate` rather than `upstream`.   Reason being: You might have a successful OS project based on this boilerplate... A contributor might then fork your project and start making contributions.  
Contributor then wants to update their fork from your repo - conventionally they would use `upstream`.

Obviously this is just a suggestion so ignore at will :D

**Comment by mxstbr** [comment]
Since there hasn't been any complaints here, as soon as CI passes I'll merge this and ship the new version!

**Comment by mxstbr** [comment]
It's shipped! Congratulations everybody, give yourself a pat on the back! Great job! https://twitter.com/mxstbr/status/794093631175069696

**Pull request merged**